internal/manifests/manifests.go

```diff
@@ -118,12 +128,20 @@ func (m *Manifests) ListClusterManifestsInternal(ctx context.Context, params ope
 	for _, file := range files {
 		parts := strings.Split(strings.Trim(file, string(filepath.Separator)), string(filepath.Separator))
 		if len(parts) > 2 {
-			manifests = append(manifests, &models.Manifest{FileName: filepath.Join(parts[3:]...), Folder: parts[2]})
+			// Is this a custom manifest?
+			fileName := filepath.Join(parts[3:]...)
+			folder := parts[2]
+			manifestMetadataPrefix := filepath.Join(params.ClusterID.String(), ManifestMetadataFolder)
+			is_custom_manifest, err := m.objectHandler.DoesObjectExist(ctx, filepath.Join(manifestMetadataPrefix, folder, fileName, "user-defined"))
```

Comment:
I dislike that we only have `user-defined` for custom manifests, and "nothing" for service created manifests.

In reality we need to make a distinction between 3 different types of manifests:
- Manifests created before this PR
- Manifests created after this PR through user upload
- Manifests created after this PR, created by the service

In the current state of this PR, it's impossible to tell apart the first and last bullets. If we also introduce a `service-defined` manifest metadata "magic" file, then we can tell apart all three bullets (both for purposes of user indication, and for the purpose of deleting service manifests during reset as suggested in one of my previous comments)

I suggest that we make the response field a non-`omitEmpty` field, this way its absence would indicate the first bullet, it's presence with `false` would indicate the second bullet, and it's presence with `true` would indicate the third bullet